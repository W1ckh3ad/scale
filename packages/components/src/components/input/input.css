.input {
  position: relative;
}
.input .input__helper-text,
.input .input__counter {
  font-weight: var(--type-weight-bold, 700);
}
.input .input__input,
.input .input__select {
  width: 100%;
  height: var(--input-large-height, 48px);
  margin: 6px 0;
  display: flex;
  outline: none;
  padding: var(--input-large-padding-y, 12px) var(--input-large-padding-x, 12px);
  z-index: 1;
  box-sizing: border-box;
  transition: var(--transition-default, all 0.2s ease-in-out);
  font-family: inherit;
  border-radius: 4px;
  border: 1px solid var(--palette-grey, #262626);
  font-size: var(--type-size-3, 1rem);
}
.input .input__textarea {
  width: 100%;
  margin: 6px 0;
  resize: vertical;
  display: flex;
  outline: none;
  padding: var(--input-large-padding-y, 12px) var(--input-large-padding-x, 12px);
  z-index: 1;
  box-sizing: border-box;
  transition: border-color 0.2s ease-in-out;
  font-family: inherit;
  border-radius: 4px;
  font-size: var(--type-size-3, 1rem);
  border: 1px solid var(--palette-grey, #262626);
}
.input .input__select {
  appearance: none;
  -webkit-appearance: none;
}
.input .input__select-wrapper {
  position: relative;
}
.input .input__counter {
  display: flex;
  transition: var(--transition-default, all 0.2s ease-in-out);
  margin-left: auto;
  padding-right: 12px;
  justify-content: flex-end;
  font-size: var(--type-variants-helper-text-font-size, 0.75rem);
  line-height: var(--type-variants-helper-text-line-height, 1.3333);
  color: var(--palette-grey, #262626);
}
.input .input__helper-text {
  transition: var(--transition-default, all 0.2s ease-in-out);
  padding-left: 12px;
  font-size: var(--type-variants-helper-text-font-size, 0.75rem);
  line-height: var(--type-variants-helper-text-line-height, 1.3333);
  color: var(--palette-helper-text, #007db3);
}
.input .input__meta {
  display: flex;
  justify-content: space-between;
}
.input:not(.input--disabled):hover .input__select-wrapper {
  --icon-color: var(--color-primary-hover, #f90984);
}
.input:not(.input--disabled):active .input__select-wrapper {
  --icon-color: var(--color-primary-active, #cb0068);
}
.input:not(.input--disabled) .input__input:hover,
.input:not(.input--disabled) .input__input:focus,
.input:not(.input--disabled) .input__select:hover,
.input:not(.input--disabled) .input__select:focus,
.input:not(.input--disabled) .input__textarea:hover,
.input:not(.input--disabled) .input__textarea:focus {
  border-color: var(--palette-magenta-hover, #f90984);
}
.input:not(.input--disabled) .input__input:focus,
.input:not(.input--disabled) .input__select:focus,
.input:not(.input--disabled) .input__textarea:focus {
  box-shadow: 0 0 0 var(--size-border-focus, 2px) var(--color-focus, #009de0);
}
.input:not(.input--disabled) .input__input:focus::placeholder,
.input:not(.input--disabled) .input__select:focus::placeholder,
.input:not(.input--disabled) .input__textarea:focus::placeholder {
  color: #6c6c6c;
  transition: var(--transition-default, all 0.2s ease-in-out);
}
.input .input__select-wrapper scale-icon {
  top: 50%;
  right: 12px;
  position: absolute;
  transform: translateY(-50%);
  pointer-events: none;
}
.input .input__input::placeholder,
.input .input__select::placeholder,
.input .input__textarea::placeholder,
.input ::placeholder {
  color: transparent;
  transition: var(--transition-default, all 0.2s ease-in-out);
}
.input--variant-static .input__label {
  color: #6c6c6c;
  display: flex;
  font-weight: 500;
}
.input--variant-animated .input__input,
.input--variant-animated .input__select {
  padding: 12px 12px 0 11px;
}
.input--variant-animated .input__textarea {
  padding-top: 24px;
}
.input--variant-animated .input__label {
  top: 0;
  left: 0;
  color: #6c6c6c;
  display: flex;
  z-index: 10;
  position: absolute;
  transition: var(--transition-default, all 0.2s ease-in-out);
  pointer-events: none;
  font-size: var(--type-size-3, 1rem);
  transform: translate(13px, calc((48px - 1rem) / 2));
  font-weight: 500;
}
.input--variant-animated.input--has-focus .input__label,
.input--variant-animated.animated .input__label {
  color: #6c6c6c;
  transform: translate(12px, 8px);
  transition: var(--transition-default, all 0.2s ease-in-out);
  font-size: 0.625rem;
  font-weight: var(--type-weight-bold, 700);
}
.input--status-error .input__input,
.input--status-error .input__textarea,
.input--status-error .input__select {
  border: 2px solid var(--palette-error, #d90000);
}
.input--status-error .input__helper-text {
  color: var(--palette-error, #d90000);
}
.input--status-error .input__counter {
  color: var(--palette-error, #d90000);
}
.input--size-small .input__input {
  height: var(--input-small-height, 40px);
}
.input--size-small .input__select {
  height: var(--input-small-height, 40px);
}
.input--size-small .input__label {
  top: 0;
  left: 0;
  color: #6c6c6c;
  display: flex;
  z-index: 10;
  position: absolute;
  transition: var(--transition-default, all 0.2s ease-in-out);
  pointer-events: none;
  font-size: var(--type-size-3, 1rem);
  transform: translate(
    13px,
    calc((var(--input-small-height, 40px) - 1rem) / 2)
  );
  font-weight: 500;
}
.input--size-small.input--has-focus .input__label,
.input--size-small.animated .input__label {
  color: #6c6c6c;
  transform: translate(12px, 4px);
  transition: var(--transition-default, all 0.2s ease-in-out);
  font-size: 0.625rem;
  font-weight: 500;
}
.input--transparent .input__input,
.input--transparent .input__textarea,
.input--transparent.input--type-radio .input__radio,
.input--transparent .input__select {
  background-color: transparent;
}
.input--type-checkbox {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
}
.input--type-checkbox input {
  width: 0;
  height: 0;
  opacity: 0;
  position: absolute;
}
.input--type-checkbox .input__meta {
  width: 100%;
}
.input--type-checkbox .input__helper-text {
  padding-left: 2rem;
}
.input--type-checkbox label {
  color: var(--palette-grey, #262626);
  font-weight: 500;
}
.input--type-checkbox .input__checkbox-container {
  width: 24px;
  height: 24px;
  display: flex;
  position: relative;
  align-items: center;
  margin-right: 0.5rem;
}
.input--type-checkbox input:disabled ~ label {
  color: var(--background-disabled, #d0d0d0);
}
.input--type-checkbox input:checked:disabled ~ label {
  color: var(--palette, #d0d0d0);
}
.input--type-checkbox
  input:checked:disabled
  ~ .input__checkbox-container
  .input__checkbox-placeholder {
  background: var(--background-disabled, #d0d0d0);
}
.input--type-checkbox
  input:checked:disabled
  ~ .input__checkbox-container
  scale-icon {
  --icon-color: #a4a4a4;
}
.input--type-checkbox
  input:checked:not([disabled]):hover
  ~ .input__checkbox-container
  .input__checkbox-placeholder,
.input--type-checkbox
  input:checked:not([disabled])
  ~ .input__checkbox-container:hover
  .input__checkbox-placeholder {
  box-shadow: none;
  border-color: var(--color-primary-hover, #f90984);
  background: var(--color-primary-hover, #f90984);
}
.input--type-checkbox
  input:checked:not([disabled]):active
  ~ .input__checkbox-container
  .input__checkbox-placeholder,
.input--type-checkbox
  input:checked:not([disabled])
  ~ .input__checkbox-container:active
  .input__checkbox-placeholder {
  border-color: #cb0068;
  background: #cb0068;
}
.input--type-checkbox
  input:checked:not([disabled]):active
  ~ .input__checkbox-container
  scale-icon,
.input--type-checkbox
  input:checked:not([disabled])
  ~ .input__checkbox-container:active
  scale-icon {
  --icon-color: #fff;
}
.input--type-checkbox
  input:checked:not([disabled])
  ~ .input__checkbox-container
  .input__checkbox-placeholder {
  border: 1px solid #e20074;
  background: #e20074;
}
.input--type-checkbox
  input:disabled
  ~ .input__checkbox-container
  .input__checkbox-placeholder {
  border-color: var(--background-disabled, #d0d0d0);
}
.input--type-checkbox
  input:focus
  ~ .input__checkbox-container
  .input__checkbox-placeholder {
  box-shadow: 0 0 0 2px #009de0;
}
.input--type-checkbox
  input:not([disabled]):hover
  ~ .input__checkbox-container
  .input__checkbox-placeholder,
.input--type-checkbox
  input:not([disabled])
  ~ .input__checkbox-container:hover
  .input__checkbox-placeholder {
  box-shadow: none;
  border-color: var(--color-primary-hover, #f90984);
}
.input--type-checkbox
  input:not([disabled]):hover
  ~ .input__checkbox-container
  ~ label,
.input--type-checkbox
  input:not([disabled])
  ~ .input__checkbox-container:hover
  ~ label {
  color: var(--color-primary-hover, #f90984);
}
.input--type-checkbox
  input:not([disabled]):active
  ~ .input__checkbox-container
  .input__checkbox-placeholder,
.input--type-checkbox
  input:not([disabled])
  ~ .input__checkbox-container:active
  .input__checkbox-placeholder {
  border-color: #cb0068;
  background: #cb0068;
}
.input--type-checkbox
  input:not([disabled]):active
  ~ .input__checkbox-container
  scale-icon,
.input--type-checkbox
  input:not([disabled])
  ~ .input__checkbox-container:active
  scale-icon {
  --icon-color: #cb0068;
}
.input--type-checkbox
  input:not([disabled]):active
  ~ .input__checkbox-container
  ~ label,
.input--type-checkbox
  input:not([disabled])
  ~ .input__checkbox-container:active
  ~ label {
  color: #cb0068;
}
.input--type-checkbox .input__checkbox-container .input__checkbox-placeholder {
  width: var(--checkbox-width, 16px);
  height: var(--checkbox-height, 16px);
  margin: 2px;
  box-sizing: border-box;
  transition: var(--transition-default, all 0.2s ease-in-out);
  margin-left: 4px;
  border-radius: 4px;
  border: 1px solid var(--palette-grey, #262626);
  background: #fff;
}
.input--type-checkbox .input__checkbox-container scale-icon {
  top: 6px;
  left: 6px;
  width: 12px;
  height: 12px;
  position: absolute;
  user-select: none;
  --icon-color: #fff;
}
.input--type-checkbox .input__checkbox-container ~ label {
  transition: var(--transition-default, all 0.2s ease-in-out);
}
.input--type-checkbox.input--status-error
  .input__checkbox-container
  .input__checkbox-placeholder {
  border: 2px solid var(--palette-error, #d90000);
}
.input--type-radio {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
}
.input--type-radio .input__meta {
  width: 100%;
}
.input--type-radio .input__helper-text {
  margin-top: 0.25rem;
  padding-left: var(--spacing-5, 1.5rem);
}
.input--type-radio label {
  color: var(--palette-grey, #262626);
  transition: var(--transition-default, all 0.2s ease-in-out);
  font-weight: 500;
}
.input--type-radio input {
  width: 16px;
  height: 16px;
  transition: var(--transition-default, all 0.2s ease-in-out);
  border-radius: 50%;
  -webkit-appearance: none;
  background-color: #fff;
  border: 1px solid var(--palette-grey, #262626);
  margin: 0 0.5rem 0 0;
}
.input--type-radio input:focus {
  outline: none;
  box-shadow: 0 0 0 2px #009de0;
}
.input--type-radio:hover input:not(:checked):not([disabled]) {
  box-shadow: none;
  border-color: var(--color-primary-hover, #f90984);
}
.input--type-radio:hover input:not(:checked):not([disabled]) ~ label {
  color: var(--color-primary-hover, #f90984);
}
.input--type-radio input:active {
  border: 8px solid #cb0068;
}
.input--type-radio input:not(:checked):not([disabled]):active ~ label {
  color: #cb0068;
}
.input--type-radio input:disabled {
  border: 1px solid var(--background-disabled, #d0d0d0);
}
.input--type-radio input:disabled ~ label {
  color: #a4a4a4;
}
.input--type-radio.input--status-error input {
  border: 2px solid var(--palette-error, #d90000);
}
.input--type-radio input:checked {
  border: 6px solid #e20074;
}
.input--type-radio input:checked:active {
  border: 6px solid #cb0068;
}
.input--type-radio input:checked:disabled {
  background: #a4a4a4;
  border: 6px solid var(--background-disabled, #d0d0d0);
}
.input--type-radio input:checked:disabled ~ label {
  color: #a4a4a4;
}
.input__textarea-label-safety-background {
  top: 2px;
  left: 2px;
  right: 2px;
  position: absolute;
  pointer-events: none;
  border-radius: var(--radius-input, 0.25rem);
  height: var(--spacing-5, 1.5rem);
  background-color: var(--background-default, #ffffff);
}
.input--disabled .input__textarea-label-safety-background {
  background-color: transparent;
}
.input--disabled label,
.input--disabled .input__label,
.input--disabled input,
.input--disabled .input__input,
.input--disabled .input__checkbox-container,
.input--disabled .input__radio,
.input--disabled .input__select,
.input--disabled .input__textarea,
.input--disabled .input__helper-text {
  cursor: not-allowed;
  border-color: var(--background-disabled, #d0d0d0);
  color: var(--background-disabled, #d0d0d0);
  background: var(--color-text-contrast, #ffffff);
}
.input--disabled .input__select-wrapper {
  --icon-color: var(--background-disabled, #d0d0d0);
}
.input--disabled.animated label.input__label {
  color: var(--background-disabled, #d0d0d0);
}
