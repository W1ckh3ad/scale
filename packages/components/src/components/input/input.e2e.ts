import { newE2EPage } from '@stencil/core/testing';

describe('scale-input', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<scale-input/>');
    const element = await page.find('scale-input');
    expect(element).toHaveClass('hydrated');
  });
});
