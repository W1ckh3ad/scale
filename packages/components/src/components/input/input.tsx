import {
  Component,
  Prop,
  Event,
  h,
  EventEmitter,
  Host,
  State,
  Watch,
  Element,
} from '@stencil/core';
import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';
import { styles } from './input.styles';
import { CssInJs } from '../../utils/css-in-js';
import { StyleSheet } from 'jss';
import Base from '../../utils/base-interface';

export interface InputChangeEventDetail {
  value: string | number | boolean | undefined | null;
}

let i = 0;

@Component({
  tag: 'scale-input',
  shadow: false,
})
export class Input implements Base {
  @Element() el: HTMLElement;
  /** (optional) Input text class */
  @Prop() customClass?: string = '';
  /** (optional) Input type */
  @Prop() type?:
    | 'email'
    | 'hidden'
    | 'number'
    | 'password'
    | 'tel'
    | 'text'
    | 'checkbox'
    | 'radio'
    | 'select'
    | 'textarea'
    | 'url' = 'text';
  /** (optional) Input name */
  @Prop() name?: string = '';
  /** (optional) Input label variant */
  @Prop() variant?: 'animated' | 'static' = 'static';
  /** (optional) Input label */
  @Prop() label: string = '';
  /** (optional) Input size */
  @Prop() size?: string = '';
  /** (optional) textarea row */
  @Prop() rows?: number;
  /** (optional) textarea column */
  @Prop() cols?: number;
  /** (optional) Input helper text */
  @Prop() helperText?: string = '';
  /** (optional) Input status */
  @Prop() status?: string = '';
  /** (optional) Input max length */
  @Prop() maxLength?: number;
  /** (optional) Input min length */
  @Prop() minLength?: number;
  /** (optional) Input placeHolder */
  @Prop() placeholder?: string = '';
  /** (optional) Input disabled */
  @Prop() disabled?: boolean;
  /** (optional) Input required */
  @Prop() required?: boolean;
  /** (optional) Input counter */
  @Prop() counter?: boolean;
  /** (optional) Active switch */
  @Prop({ reflect: true }) checked?: boolean = false;
  /** (optional) textarea resize */
  @Prop() resize?: 'unset' | 'none' | 'vertical' | 'horizontal';
  /** (optional) Input value */
  @Prop({ mutable: true }) value?: string | number | null = '';
  /** (optional) Input checkbox id */
  @Prop() inputId?: string;
  /** (optional) Input checkbox checked icon */
  @Prop() icon?: string;
  /** (optional) select multiple options */
  @Prop() multiple?: boolean;
  /** (optional) the number of visible options in a select drop-down list */
  @Prop() visibleSize?: number;
  /** (optional) input background transparent */
  @Prop() transparent?: boolean;

  /** (optional) Injected jss styles */
  @Prop() styles?: any;
  /** decorator Jss stylesheet */
  @CssInJs('Input', styles) stylesheet: StyleSheet;

  /** Emitted when a keyboard input occurred. */
  @Event() scaleInput!: EventEmitter<KeyboardEvent>;
  /** Emitted when the value has changed. */
  @Event() scaleChange!: EventEmitter<InputChangeEventDetail>;
  /** Emitted when the input has focus. */
  @Event() scaleFocus!: EventEmitter<void>;
  /** Emitted when the input loses focus. */
  @Event() scaleBlur!: EventEmitter<void>;
  /** Emitted on keydown. */
  @Event() scaleKeyDown!: EventEmitter<KeyboardEvent>;

  /** (optional) Input checkbox checked */
  @State() customResize?: any;

  componentWillLoad() {
    if (this.inputId == null) {
      this.inputId = 'input-' + i++;
    }
  }

  componentWillUpdate() {}
  componentDidLoad() {
    if (this.type === 'select') {
      const select = this.el.querySelector('select');
      const selectedValue = select.options[select.selectedIndex].value;
      this.value = selectedValue;
    }
  }
  componentDidUnload() {}

  @Watch('value')
  valueChanged() {
    this.scaleChange.emit({
      value: this.value == null ? this.value : this.value.toString(),
    });
  }

  checkedChanged = event => {
    if (!this.disabled) {
      this.scaleChange.emit(event);
    }
  };

  handleInput = (event: Event) => {
    const target = event.target as HTMLInputElement | null;
    if (target) {
      this.value = target.value || '';
    }
    this.scaleInput.emit(event as KeyboardEvent);
  };

  handleChange = (event: Event) => {
    const target = event.target as HTMLInputElement | null;
    if (target) {
      this.value = target.value || '';
    }
  };

  handleFocus = () => {
    this.scaleFocus.emit();
  };

  handleBlur = () => {
    this.scaleBlur.emit();
  };

  handleKeyDown = (event: KeyboardEvent) => {
    this.scaleKeyDown.emit(event);
  };

  render() {
    if (this.type === 'checkbox') {
      return (
        <Host checked={this.checked}>
          <div class={this.getCssClassMap()}>
            <input
              type="checkbox"
              name={this.name}
              id={this.inputId}
              checked={this.checked}
              disabled={this.disabled}
              onClick={this.checkedChanged}
            />
            <div
              class={classNames('input__checkbox-container')}
              onClick={this.checkedChanged}
            >
              <span class={classNames('input__checkbox-placeholder')}></span>
              {!!this.icon && (
                <scale-icon path={this.icon} size={12}></scale-icon>
              )}
            </div>
            <label htmlFor={this.inputId}>{this.label}</label>
          </div>
        </Host>
      );
    }

    if (this.type === 'radio') {
      return (
        <Host>
          <div class={this.getCssClassMap()}>
            <input
              type="radio"
              name={this.name}
              id={this.inputId}
              onChange={this.checkedChanged}
              value={this.value}
              checked={this.checked}
              disabled={this.disabled}
            />
            <label htmlFor={this.inputId}>{this.label}</label>
          </div>
        </Host>
      );
    }
    const Tag = this.type === 'textarea' ? 'textarea' : 'input';

    return (
      <Host>
        <div class={this.getCssClassMap()}>
          {this.variant === 'static' && (
            <label class="input__label" htmlFor={this.inputId}>
              {this.label}
            </label>
          )}
          {this.type === 'select' ? (
            <div class="input__select-wrapper">
              <select
                class={classNames('input__select')}
                onChange={this.handleChange}
                onFocus={this.handleFocus}
                onBlur={this.handleBlur}
                onKeyDown={this.handleKeyDown}
                disabled={this.disabled}
                required={this.required}
                multiple={this.multiple}
                id={this.inputId}
                size={this.visibleSize}
              >
                <slot />
              </select>
              {!!this.icon && <scale-icon path={this.icon}></scale-icon>}
            </div>
          ) : (
            <Tag
              type={this.type}
              class={classNames(
                `input__${this.type === 'textarea' ? 'textarea' : 'input'}`,
                this.customResize && this.customResize.id
              )}
              style={!!this.resize && { resize: this.resize }}
              value={this.value}
              {...(!!this.name ? { name: this.name } : {})}
              required={this.required}
              minLength={this.minLength}
              maxLength={this.maxLength}
              id={this.inputId}
              onInput={this.handleInput}
              onChange={this.handleChange}
              onFocus={this.handleFocus}
              onBlur={this.handleBlur}
              onKeyDown={this.handleKeyDown}
              {...(!!this.placeholder ? { placeholder: this.placeholder } : {})}
              disabled={this.disabled}
              {...(!!this.rows ? { rows: this.rows } : {})}
              {...(!!this.cols ? { cols: this.cols } : {})}
            />
          )}

          {this.variant === 'animated' && (
            <label class="input__label" htmlFor={this.inputId}>
              {this.label}
            </label>
          )}
          {(!!this.helperText || !!this.counter) && (
            <div class="input__meta">
              {!!this.helperText && (
                <div class="input__helper-text">{this.helperText}</div>
              )}
              {this.counter && (
                <div class="input__counter">
                  {!!this.value ? String(this.value).length : 0} /{' '}
                  {this.maxLength}
                </div>
              )}
            </div>
          )}
        </div>
      </Host>
    );
  }

  getCssClassMap(): CssClassMap {
    const { classes } = this.stylesheet;

    return classNames(
      classes.input,
      this.customClass && this.customClass,
      this.type && classes[`input--type-${this.type}`],
      this.checked && classes[`input--checked`],
      this.resize && classes[`input--resize-${this.resize}`],
      this.variant && classes[`input--variant-${this.variant}`],
      this.disabled && classes[`input--disabled`],
      this.transparent && classes['input--transparent'],
      this.status && classes[`input--status-${this.status}`],
      this.size && classes[`input--size-${this.size}`],
      this.value && 'animated'
    );
  }
}
