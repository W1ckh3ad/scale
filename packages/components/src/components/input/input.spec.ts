import { newSpecPage } from '@stencil/core/testing';
import { Input } from './input';
import { styles } from './input.styles';
import jss from 'jss';

describe('Input', () => {
  let element;
  let stylesheet;
  beforeEach(async () => {
    element = new Input();
    stylesheet = element.stylesheet = jss.createStyleSheet(styles as any);
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Input],
      html: `<scale-input></scale-input>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle css classes', () => {
    element.customClass = 'custom';
    expect(element.getCssClassMap()).toContain('custom');

    element.type = 'number';
    stylesheet.addRule('input--type-number', {});
    expect(element.getCssClassMap()).toContain(
      stylesheet.classes['input--type-number']
    );
  });
});
