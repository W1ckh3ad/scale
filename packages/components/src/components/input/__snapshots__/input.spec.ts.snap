// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot 1`] = `
<scale-input>
  <!---->
  <style>
    .input-0-0-10 {
  position: relative;
}
.input-0-0-10 .input__helper-text, .input-0-0-10 .input__counter {
  font-family: TeleNeoWeb-Bold;
}
.input-0-0-10 .input__input, .input-0-0-10 .input__select {
  width: 100%;
  border: 1px solid #cecece;
  height: 48px;
  margin: 6px 0;
  display: flex;
  padding: 12px 12px;
  z-index: 1;
  font-size: 16px;
  box-sizing: border-box;
  transition: all 0.2s ease-in-out;
  font-family: inherit;
  border-radius: 4px;
}
.input-0-0-10 .input__textarea {
  border: 1px solid #cecece;
  margin: 6px 0;
  display: flex;
  padding: 12px 12px;
  z-index: 1;
  font-size: 16px;
  box-sizing: border-box;
  transition: border-color 0.2s ease-in-out;
  font-family: inherit;
  border-radius: 4px;
}
.input-0-0-10 .input__select {
  appearance: none;
}
.input-0-0-10 .input__select-wrapper {
  position: relative;
}
.input-0-0-10 .input__counter {
  color: #333;
  display: flex;
  font-size: 12px;
  transition: all 0.2s ease-in-out;
  margin-left: auto;
  justify-content: flex-end;
}
.input-0-0-10 .input__helper-text {
  font-size: 12px;
  transition: all 0.2s ease-in-out;
}
.input-0-0-10 .input__meta {
  display: flex;
  justify-content: space-between;
}
.input-0-0-10:not(.input--disabled-0-0-15) {}
.input-0-0-10:not(.input--disabled-0-0-15) .input__input, .input-0-0-10:not(.input--disabled-0-0-15) .input__select, .input-0-0-10:not(.input--disabled-0-0-15) .input__textarea {}
.input-0-0-10:not(.input--disabled-0-0-15) .input__checkbox-container {}
.input-0-0-10:not(.input--disabled-0-0-15) .input__radio {}
.input-0-0-10:not(.input--disabled-0-0-15) .input__radio:hover, .input-0-0-10:not(.input--disabled-0-0-15) .input__radio:focus {
  border-color: #e20074;
}
.input-0-0-10:not(.input--disabled-0-0-15) .input__checkbox-container:hover, .input-0-0-10:not(.input--disabled-0-0-15) .input__checkbox-container:focus {}
.input-0-0-10:not(.input--disabled-0-0-15) .input__checkbox-container:hover .input__checkbox-placeholder, .input-0-0-10:not(.input--disabled-0-0-15) .input__checkbox-container:focus .input__checkbox-placeholder {
  border-color: #e20074;
}
.input-0-0-10:not(.input--disabled-0-0-15) .input__input:hover, .input-0-0-10:not(.input--disabled-0-0-15) .input__input:focus, .input-0-0-10:not(.input--disabled-0-0-15) .input__select:hover, .input-0-0-10:not(.input--disabled-0-0-15) .input__select:focus, .input-0-0-10:not(.input--disabled-0-0-15) .input__textarea:hover, .input-0-0-10:not(.input--disabled-0-0-15) .input__textarea:focus {
  border-color: #e20074;
}
.input-0-0-10 .input__select-wrapper scale-icon {
  top: 50%;
  right: 0;
  position: absolute;
  transform: translateY(-50%);
}
.input--variant-static-0-0-11 {}
.input--variant-static-0-0-11 .input__label {
  display: flex;
}
.input--variant-animated-0-0-12 {}
.input--variant-animated-0-0-12 .input__input:focus + .input__label, .input--variant-animated-0-0-12.animated .input__label {
  font-size: 10px;
  transform: translate(12px, 8px);
  transition: all 0.2s ease-in-out;
  font-family: TeleNeoWeb-Bold;
  font-weight: 700;
  color: #e20074;
}
.input--variant-animated-0-0-12 .input__input, .input--variant-animated-0-0-12 .input__select {
  padding: 12px 12px 0 11px;
}
.input--variant-animated-0-0-12 .input__textarea {
  padding-top: 24px;
}
.input--variant-animated-0-0-12 .input__label {
  top: 0;
  left: 0;
  color: #6c6c6c;
  display: flex;
  z-index: 10;
  position: absolute;
  font-size: 16px;
  transform: translate(13px, 16px);
  transition: all 0.2s ease-in-out;
  font-weight: 400;
  pointer-events: none;
}
.input--variant-animated-0-0-12 .input__input:focus + .input__label, .input--variant-animated-0-0-12 .input__select:focus + .input__label, .input--variant-animated-0-0-12.animated .input__label {
  font-size: 10px;
  transform: translate(12px, 8px);
  transition: all 0.2s ease-in-out;
  font-weight: 700;
  color: #e20074;
}
.input--variant-animated-0-0-12 .input__textarea:focus + .input__label, .input--variant-animated-0-0-12.animated .input__label {
  font-size: 10px;
  transform: translate(12px, 8px);
  transition: all 0.2s ease-in-out;
  font-family: TeleNeoWeb-Bold;
  font-weight: 700;
  color: #e20074;
}
.input--status-error-0-0-13 {}
.input--status-error-0-0-13.animated .input__label, .input--status-error-0-0-13 label {
  color: #f50057 !important;
}
.input--status-error-0-0-13 .input__input:focus + .input__label, .input--status-error-0-0-13 .input__textarea:focus + .input__label {
  color: #f50057 !important;
}
.input--status-error-0-0-13 .input__input:focus + .input__label {
  color: #f50057 !important;
}
.input--status-error-0-0-13 .input__input {
  border: 1px solid #f50057;
}
.input--status-error-0-0-13 .input__helper-text {
  color: #f50057;
}
.input--status-error-0-0-13 .input__counter {
  color: #f50057;
}
.input--size-small-0-0-14 {}
.input--size-small-0-0-14 .input__input {
  height: 40px;
}
.input--size-small-0-0-14 .input__label {
  top: 0;
  left: 0;
  color: #6c6c6c;
  display: flex;
  z-index: 10;
  position: absolute;
  font-size: 16px;
  transform: translate(13px, 12px);
  transition: all 0.2s ease-in-out;
  font-weight: 400;
  pointer-events: none;
}
.input--size-small-0-0-14 .input__input:focus + .input__label, .input--size-small-0-0-14.animated .input__label {
  font-size: 10px;
  transform: translate(12px, 4px);
  transition: all 0.2s ease-in-out;
  font-weight: 700;
  color: #e20074;
}
.input--disabled-0-0-15 {}
.input--disabled-0-0-15 .input__label, .input--disabled-0-0-15 .input__input, .input--disabled-0-0-15 .input__checkbox-container, .input--disabled-0-0-15 .input__radio, .input--disabled-0-0-15 .input__select, .input--disabled-0-0-15 .input__textarea {
  cursor: not-allowed!important;
  opacity: 0.5;
}
.input--type-checkbox-0-0-16 {
  display: flex;
  align-items: center;
}
.input--type-checkbox-0-0-16 .input__checkbox-container {
  width: 24px;
  height: 24px;
  display: flex;
  position: relative;
  align-items: center;
  margin-right: 10px;
}
.input--type-checkbox-0-0-16:not(.input--disabled-0-0-15) {}
.input--type-checkbox-0-0-16:not(.input--disabled-0-0-15) .input__checkbox-container {}
.input--type-checkbox-0-0-16:not(.input--disabled-0-0-15) .input__checkbox-container .input__checkbox-placeholder {
  border: 1px solid #111111;
  background-color: #D8D8D8;
}
.input--type-checkbox-0-0-16:not(.input--disabled-0-0-15) .input__checkbox-container:hover {}
.input--type-checkbox-0-0-16:not(.input--disabled-0-0-15) .input__checkbox-container:hover .input__checkbox-placeholder {
  border: 1px solid #E20074;
  background-color: #ffffff;
}
.input--type-checkbox-0-0-16 .input__checkbox-container .input__checkbox {
  top: 0;
  left: 0;
  width: 100%;
  cursor: inherit;
  height: 100%;
  margin: 0;
  opacity: 0;
  padding: 0;
  z-index: 1;
  position: absolute;
}
.input--type-checkbox-0-0-16 .input__checkbox-container .input__checkbox-placeholder {
  width: 16px;
  border: 1px solid #cecece;
  height: 16px;
  margin: 2px;
  border-radius: 4px;
}
.input--checked-0-0-17 {}
.input--checked-0-0-17 scale-icon {
  top: 6px;
  left: 6px;
  width: 12px;
  height: 12px;
  position: absolute;
  user-select: none;
}
.input--checked-0-0-17.input--disabled-0-0-15 {}
.input--checked-0-0-17:not(.input--disabled-0-0-15) {}
.input--checked-0-0-17:not(.input--disabled-0-0-15) .input__checkbox-container {}
.input--checked-0-0-17:not(.input--disabled-0-0-15) scale-icon {
  --icon-color: #ffffff;
}
.input--checked-0-0-17:not(.input--disabled-0-0-15) .input__checkbox-container .input__checkbox-placeholder {
  border: 1px solid #E20074;
  background-color: #E20074;
}
.input--checked-0-0-17:not(.input--disabled-0-0-15) .input__checkbox-container:hover {}
.input--checked-0-0-17:not(.input--disabled-0-0-15) .input__checkbox-container:hover .input__checkbox-placeholder {
  border: 1px solid #E20074;
  background-color: #F90984;
}
.input--checked-0-0-17.input--disabled-0-0-15 .input__checkbox-placeholder {
  background-color: #cdcdcd;
}
.input--type-radio-0-0-18 {
  display: flex;
  align-items: center;
}
.input--type-radio-0-0-18 .input__radio {
  width: 16px;
  border: 1px solid #cecece;
  height: 16px;
  appearance: none;
  margin-right: 10px;
  border-radius: 50%;
  background-color: #ffffff;
}
.input--type-radio-0-0-18.input--disabled-0-0-15 {}
.input--type-radio-0-0-18.input--disabled-0-0-15 .input__radio {}
.input--type-radio-0-0-18.input--disabled-0-0-15 .input__radio:checked {
  border: 6px solid #cdcdcd;
  background-color: #979797;
}
.input--type-radio-0-0-18 .input__radio:checked, .input--type-radio-0-0-18 .input__radio:focus {
  outline: none;
  transition: all 0.2s ease-in-out;
  border: 6px solid #e20074;
}
  </style>
  <div class="input--variant-static-0-0-11 input-0-0-10">
    <label class="input__label"></label>
    <input class="input__input" type="text">
  </div>
</scale-input>
`;
